### PR TITLE
Add support for two digit versions - vX.Y

### LINKED ISSUE #24: Two Digit Versions
Hi All, 

While SemVer requires X.Y.Z it would be great to support X.Y as well as this is also a common version format used in GitHub.

To illustrate:
`fetch --repo="https://github.com/dnephin/dobi" --tag=">=0.9" --release-asset="dobi-windows.exe" %CMDER_ROOT%\bin
`
Fails, while
`fetch --repo="https://github.com/dnephin/dobi" --tag="v0.9" --release-asset="dobi-windows.exe" %CMDER_ROOT%\bin
`
Works but the 'v' is required to make this work.

### PR DESCRIPTION
**Pull request by tomdee**
The version parsing library already supports them, but it normalizes
all version strings to have three components. This fix properly handles
versions that have been normalized in that way.

Fixes #24

### CODE REVIEW

File: tag.go [hunk 1]
```diff
@@ -74,9 +74,12 @@ func getLatestAcceptableTag(tagConstraint string, tags []string) (string, *Fetch
 		return latestTag, wrapError(errors.New("Tag does not exist"))
 	}
 
-	// The tag name may have started with a "v". If so, re-apply that string now
+	// go-version normalizes versions to remove the leading v and uses at least three version segments x.y.z
+	// find the origina version string that go-version matched
 	for _, originalTagName := range tags {
-		if strings.Contains(originalTagName, latestAcceptableVersion.String()) {
+		// ignore any errors since the tag was previously successfully converted
+		originalTagNameConverted, _ := version.NewVersion(originalTagName)
```
Comment (brikis98) [inline tag.go]: Ignoring errors like this seems very error prone. Some future refactor of the code may break the assumption you're making in the comment. What about saving the output of the `version.NewVersion` calls above into a list and looping over that list here?

**Comment by tomdee** [comment]
@brikis98 I've addressed your comment and added some test cases. I've just amended my previous commit as this change is quite small, PTAL.

File: tag_test.go [hunk 2]
```diff
@@ -19,12 +19,15 @@ func TestGetLatestAcceptableTag(t *testing.T) {
...
 		{"~> 1.1.1", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0"}, "1.1.3"},
 		{"~> 1.2.1", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0"}, "1.2.3"},
-		{"~> 1.1", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0"}, "1.4.0"},
-		{"~> 1.2", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0"}, "1.4.0"},
+		{"~> 1.1", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0", "1.3", "2.3"}, "1.4.0"},
+		{"~> 1.2", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0", "1.3", "2.3"}, "1.4.0"},
 		{"~> 1.3", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0"}, "1.4.0"},
+		{"~> 1.3", []string{"1.0.5","1.0.6","1.0.7","1.0.8","1.0.9","1.1.0","1.1.1","1.1.2","1.1.3","1.2.3","1.4.0","2.0.0","2.1.0", "1.5"}, "1.5"},
```
Comment (josh-padnick) [inline tag_test.go]: Could you just add a test like the following:

```golang
{"~> 1.3", []string{"1.2", "1.3", "1.3.1"}, "1.3.1"}, 
```

I'd like to make sure that even when a 2-digit version satisfies the constraint, we still select the latest version, even if it's a 3-digit version.

**Comment by tomdee** [comment]
Test added, back to you.

**Comment by brikis98** [comment]
I'll merge now and let the tests run. Once they pass, I'll issue a new release.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/fetch/releases/tag/v0.3.1. Thx @tomdee!

**Pull request merged**